backend/pcloud/pcloud.go

```diff
@@ -1033,6 +1036,224 @@ func (f *Fs) Shutdown(ctx context.Context) error {
 	return nil
 }
 
+// changeNotify implements fs.Features.ChangeNotify
+func (f *Fs) changeNotify(ctx context.Context, notify func(string, fs.EntryType), ch <-chan time.Duration) {
+	// Start long-poll loop in background
+	go f.changeNotifyLoop(ctx, notify, ch)
+}
+
+// changeNotifyLoop contains the blocking long-poll logic.
+func (f *Fs) changeNotifyLoop(ctx context.Context, notify func(string, fs.EntryType), ch <-chan time.Duration) {
+	// Standard polling interval
+	interval := 30 * time.Second
+
+	// Start with diffID = 0 to get the current state
+	var diffID int64 = 0
+
+	// Cache to map FolderID (int64) -> Rclone Path (string)
+	// This local cache is essential because the API provides events by ID,
+	// but rclone needs paths. We cannot rely on rclone's internal dirCache
+	// here as it is optimized for Path->ID, not ID->Path.
+	folderPathCache := make(map[int64]string)
+	folderPathCache[0] = "" // Initialize with Root
+
+	// Safety: Limit cache size to prevent memory leaks on long-running mounts
+	const maxCacheSize = 10000
+
+	// Forward declaration for recursion
+	var resolvePath func(int64) string
+
+	// Helper: Recursively resolve pCloud folder ID to rclone relative path
```

Comment:
Factor this helper into a new method on Fs as it is a bit long.

It would be better to look in the dirCache first as it may have the desired result (use the `GetInv` method). This should also update the dirCache with any results it finds (use `Put` method) - there is absolutely no point keeping two caches.

So change it to use the dirCache. If something is found by ID then use it, if not then use the code here.